File: fsharp/log-levels/LogLevels.fs
module LogLevels

let message (logLine: string): string = (logLine.Split(':')[1]).Trim()

let logLevel(logLine: string): string =
    let idx = logLine.IndexOf(':')
    logLine.Substring(1, idx - 2).ToLower()

let reformat(logLine: string): string =
    let msg = message logLine
    let loglvl = logLevel logLine
    sprintf "%s (%s)" msg loglvl

File: java/log-levels/src/main/java/LogLevels.java
public class LogLevels {

    public static String message(String logLine) {
        return logLine.split("]:")[1].strip();
    }

    public static String logLevel(String logLine) {
        int beg = logLine.indexOf("[") + 1;
        int end = logLine.indexOf("]");
        return logLine.substring(beg, end).toLowerCase();
    }

    public static String reformat(String logLine) {
        String mess = message(logLine);
        String levl = logLevel(logLine);
        return mess + " (" + levl + ")";
    }
}
